HTTPServer.test/ServerTests.cs

```diff
@@ -21,23 +25,39 @@ public class ServerUnitTests
         [TestMethod]
         public void ServerCanStart()
         {
-            Assert.True(server.Start() && server.Running);
+            Assert.True(server.Start(8080) != null && server.Running);
             server.Stop();
         }
 
         [TestMethod]
         public void ServerCanStop()
         {
-            server.Start();
+            server.Start(8080);
             Assert.True(server.Stop() && !server.Running);
         }
 
         [TestMethod]
         public void ServerCanEchoData()
         {
             SetUpClient();
-            ConnectClientToServer(socket, ipEndPoint);
-            var message = CommunicateWithServer(socket, bytesReturned);
+            var message = "";
+            Action<object> action1 = (object obj) =>
+            {
+                ConnectClientToServer(socket, ipEndPoint);
+            };
+
+            Action<object> action2 = (object obj) =>
+            {
+                message = CommunicateWithServer(socket, bytesReturned);
+            };
+
+            Task t1 = new Task(action1, "");
+            Task t2 = new Task(action2,"");
+
+            t1.Start();
+            System.Threading.Thread.Sleep(5000);
```

Comment:
Is there a way to add a callback on the first task which executes the second task?